### PR TITLE
ENH: Increase coverage for `itk::SobelOperator`

### PR DESCRIPTION
**Pull request by jhlegarreta**
Increase coverage for `itk::SobelOperator`:
- Exercise basic object methods using the `ITK_EXERCISE_BASIC_OBJECT_METHODS` macro.
- Test the Set/Get methods using the `ITK_TEST_SET_GET_VALUE` macro.

Test the operator using fundametal convolutional operations and using the `itk::NeighborhoodOperatorImageFilter` filter: ensure that both yield the same result.

Add the accompanying content data link files.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Added test (or behavior not changed)
- [X] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
This tests the operator quantitatively; the existing good coverage is being achieved through this test:
https://github.com/InsightSoftwareConsortium/ITK/blob/23da3dfa79152648f3d38050ee147df47f41ce0a/Modules/Core/Common/test/itkNeighborhoodOperatorTest.cxx#L208

However, it is not quantitative.

I used the [`sf4.png`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Testing/Data/Input/sf4.png.sha512) image for the test; [`chondt.png`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Testing/Data/Input/chondt.png.sha512) is also a good candidate to show the filter. Maybe the latter can be used to demonstrate the operator in the ITKSphinxExamples.

**Comment by jhlegarreta** [comment]
No idea why the Linux builds are failing :shrug::
https://dev.azure.com/itkrobotlinux/ITK.Linux/_build/results?buildId=9158&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=f58be928-45a1-58e7-c1ad-04869565b3f4&l=94

**Review by thewtex — APPROVED** [review]
@jhlegarreta awesome, thank you! :green_heart:

File: Modules/Core/Common/test/itkSobelOperatorImageConvolutionTest.cxx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  SobelOperatorType sobelOperator;
+
+  auto direction = std::stoul(argv[2]);
+  sobelOperator.SetDirection(direction);
+
+  itk::Size<Dimension> radius;
+  radius.Fill(1);
+  sobelOperator.CreateToRadius(radius);
+
+  NeighborhoodIteratorType it(radius, inputImage, inputImage->GetRequestedRegion());
```
Comment (issakomi) [inline Modules/Core/Common/test/itkSobelOperatorImageConvolutionTest.cxx:62]: Maybe change 'requested region' to 'largest possible region' here and below? This test randomly fails, e.g.
https://open.cdash.org/test/862523065
https://open.cdash.org/test/862546031
https://open.cdash.org/test/862426223
and many more.
Not tested and not 100% sure.
cc @dzenanz

Comment (jhlegarreta) [reply Modules/Core/Common/test/itkSobelOperatorImageConvolutionTest.cxx:62]: Thanks for reporting @issakomi :+1:.

Not sure if the region might be the cause. I realize that I used the same names for the output files of the two different tests `itkSobelOperatorImageFilterTest` and `itkSobelOperatorImageConvolutionTest`: maybe the issue is that tests clash with each other. Despite the reported failures only related to the horizontal detector. I'll propose a patch set to use different output names for each test and see if that solves it.

Comment (jhlegarreta) [reply Modules/Core/Common/test/itkSobelOperatorImageConvolutionTest.cxx:62]: PR #3746. Resolving.

**Pull request merged**